src/wms/src/main/java/org/geoserver/wms/GetMap.java

```diff
@@ -666,6 +666,18 @@ WebMap executeInternal(
                 WMSLayerInfo wmsLayer = (WMSLayerInfo) mapLayerInfo.getResource();
                 WebMapServer wms = wmsLayer.getStore().getWebMapServer(null);
                 Layer gt2Layer = wmsLayer.getWMSLayer(null);
+                if (wmsLayer.isRespectMetadataBBox()) {
+                    boolean isInsideBounnds =
+                            wmsLayer.checkEnvelopOverLapWithNativeBounds(
+                                    mapContent.getViewport().getBounds());
+                    if (!isInsideBounnds) {
+                        LOGGER.warning(
```

Comment:
Definitely not something that should show up as warning, it's a normal behavior requested by the admin. I suggest replacing with a "FINE" log and guard it with a test on whether the logging level allows for FINE logging, or switch to a log with string + parameters (so that no string concatenation happens anyways).

Reply:
understood will do